mantidproject/mantid
# Remove TOF axis adjustment from LoadILLTOF

This PR updates the ILL NeXus loader for IN4, IN5 and IN6 instruments. The new loader loads the time-of-flight axis as specified in the NeXus file, without trying to adjust the first bin boundary with regards to the elastic channel. Consequently, the vanadium reference input properties were removed as well.

The old loader is marked as deprecated.

**To test:**

There are NeXus files available for all the supported instruments in
`<mantid-build-dir>/ExternalData/Testing/Data/UnitTest/ILL`

Both loader version 1 and version 2 should produce exactly the same content, except the starting bin boundaries in version 2 are taken from the 'monitor.time_of_flight' sample logs while in version 1 the bin boundaries are shifted. Also, the 'EPP' entry is missing in the sample logs in files loaded by version 2.

The following script could be used for some testing:
```python
import numpy

SAMPLE_LOGS = 'duration,Detector.elasticpeak,end_time,Ei,wavelength,experiment_identifier,Facility,run_number,start_time,title'

def checkSpectra(ws1, ws2):
    '''
    Compares workspaces loaded with LoadILLTOF v1 and 2.
    '''
    # X bin widths should be equal.
    xs1 = ws1.extractX()
    xs2 = ws2.extractX()
    diffxs1 = numpy.diff(xs1)
    diffxs2 = numpy.diff(xs2)
    result = numpy.all(diffxs1 == diffxs2)
    print('X test, ws1: {} ws2: {} match? {}'.format(str(ws1), str(ws2), result))
    # DX doesn't exists here, but check it anyway.
    dxs1 = ws1.extractDx()
    dxs2 = ws2.extractDx()
    result = numpy.all(dxs1 == dxs2)
    print('DX test, ws1: {} ws2: {} match? {}'.format(str(ws1), str(ws2), result))
    # Counts should be equal.
    ys1 = ws1.extractY()
    ys2 = ws2.extractY()
    result = numpy.all(ys1 == ys2)
    print('Y test, ws1: {} ws2: {} match? {}'.format(str(ws1), str(ws2), result))
    # Count errors should be equal.
    es1 = ws1.extractE()
    es2 = ws2.extractE()
    result = numpy.all(es1 == es2)
    print('E test, ws1: {} ws2: {} match? {}'.format(str(ws1), str(ws2), result))
    # See if some random sample logs are the same.
    result = CompareSampleLogs('{}, {}'.format(str(ws1), str(ws2)), SampleLogs=SAMPLE_LOGS)
    # result contains the names of the logs which differ.
    print('Sample log comparison: {}'.format(result))


in4_file = 'ADD-FILE-NAME-HERE'

in4_v1 = LoadILLTOF(Filename=in4_file, Version=1)
in4_v2 = LoadILLTOF(Filename=in4_file, Version=2)

checkSpectra(in4_v1, in4_v2)
```


Fixes #18091 
The other piece needed to fix the issue is handled in this PR: #18118 

*The release notes were updated accordingly.*


---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
